### DISCUSSION TITLE
Setting the default PKCE preference using AspNet integration

### DISCUSSION THREAD

**Discussion by kurabirko** [scalar/scalar#4156] [Q&A] [answered]
Hi all

I have an AspNet minimal api project where I use Scalar with an OAuth2 identity provider. Well, I use Keycloak with OIDC but I don't think this should make a difference in this case. After defining the OAuth flow in my OpenApi document and registering it to scalar i can authenticate just fine. However, I have to manually enter my PKCE preference each time I authenticate. Is there an option to set this default somewhere in the configuration api? If not maybe that might be a nice addition and if it is relatively simple I can create a quick PR.

**Comment by xC0dex** [comment, marked as answer]
Hey @kurabirko,
I didn't verify this, but I think Scalar supports the [`x-use-pkce`](https://redocly.com/docs-legacy/api-reference-docs/specification-extensions/x-use-pkce) extension. So I guess you could add this extension to your security scheme using a custom transformer.

**Reply by kurabirko** [reply]
I just had some time to test this, so my reply is coming a bit late. `x-use-pkce` indeed works (see #3705), so I will be using it until the asp.net support lands. Thanks!

**Comment by AndreiGorlov** [comment]
https://github.com/scalar/scalar/issues/604#issuecomment-2456381787
